src/java.base/share/classes/java/lang/foreign/FunctionDescriptor.java

```diff
@@ -94,6 +95,22 @@ public sealed interface FunctionDescriptor permits FunctionDescriptorImpl {
      */
     FunctionDescriptor dropReturnLayout();
 
+    /**
+     * Returns the method type consisting of the carrier types of the layouts in this function descriptor.
+     * <p>
+     * The carrier type of a layout is determined as follows:
+     * <ul>
+     * <li>If the layout is a {@link ValueLayout} the carrier type is determined through {@link ValueLayout#carrier()}.</li>
+     * <li>If the layout is a {@link GroupLayout}, or {@link SequenceLayout} the carrier type is {@link MemorySegment}.</li>
+     * <li>If the layout is a {@link PaddingLayout} an {@link IllegalArgumentException} is thrown.</li>
```

Comment:
This comment seems to be inconsistent with the @throws clause

Reply:
The implementation seems to support sequence layouts, and return MemorySegment. Not super sure about that (given the linker doesn't support that).

Reply:
In the interest of moving this forward, for now I'll keep the current mapping behavior from Linker.methodType, and just do the move to FunctionDescriptor in this patch.

Reply:
Note that even if we keep behavior, the javadoc `@throws` seems still inconsistent with what the method javadoc says.